### PR TITLE
Add stock-location-pi endpoint

### PR DESCRIPTION
**Pull request by parisiale**

### CODE REVIEW

File: controllers/stock_location.py [hunk 1]
```diff
@@ -1,14 +1,41 @@
 # -*- coding: utf-8 -*-
 
+import json
+import os
+
+import jsonschema
```
Comment (parisiale) [inline controllers/stock_location.py]: New dependency.

File: README.md [hunk 2]
```diff
@@ -334,6 +335,16 @@ Params:
...
 ```
 
+### Stock Location PI Count
+```
+URI: /api/stock-location-pi-count/
+Method: POST
+Params:
+@param pi_request: JSON object with "pi_count_moves", "inventory_adjustments", "preceding_inventory_adjustments" and "location_id" entries
+```
+
```
Comment (ws-mps) [inline README.md:347]: Can we add a brief description of what it is going to do?

File: controllers/stock_location.py [hunk 3]
```diff
@@ -27,11 +29,45 @@ def get_location(self, location_id=None, location_name=None, location_barcode=No
...
+        try:
+            location_id = int(pi_request.get('location_id'))
+        except ValueError:
+            pass
+
+        if location_id is None:
+            raise ValidationError(
+                _('You need to provide a valid id for the location.'))
+
+        location = Location.browse(location_id)
```
Comment (ws-mps) [inline controllers/stock_location.py]: We can use `Location.get_location(location_id)`, then lines 70 and 71 are not needed.

File: README.md (hunk 2 above)
Comment (parisiale) [reply README.md:347]: Done.

File: models/stock_location.py [hunk 4]
```diff
@@ -19,3 +19,18 @@ def _prepare_info(self, extended=False, **kwargs):
...
+    def _process_pi_count_moves(self, count_moves_request,
+                                picking_type_id=None):
+        """
+            Invokes the parent class method by specifying the
+            picking type id of the current user's warehouse.
+        """
+        Users = self.env['res.users']
+
+        warehouse = Users.get_user_warehouse()
+        picking_type_id = warehouse.pick_type_id.id
```
Comment (ws-mps) [inline models/stock_location.py]: This is not the picking type you are looking for

**Pull request merged**